packages/datadog-instrumentations/src/express.js

```diff
@@ -129,7 +145,37 @@ addHook({ name: 'express', versions: ['>=4.0.0 <4.3.0'] }, express => {
   return express
 })
 
-addHook({ name: 'express', versions: ['>=4.3.0'] }, express => {
+addHook({ name: 'express', versions: ['>=4.3.0 <5.0.0'] }, express => {
   shimmer.wrap(express.Router, 'process_params', wrapProcessParamsMethod(2))
   return express
 })
+
+addHook({ name: 'express', file: ['lib/express.js'], versions: ['>=5.0.0'] }, express => {
+  shimmer.wrapFunction(express, function (original) {
+    const app = original.call(this, arguments)
+    const requestProto = Object.getPrototypeOf(app.request)
+    const requestDescriptor = Object.getOwnPropertyDescriptor(requestProto, 'query')
```

Comment:
I don't think this gonna work, because we need to execute createApplication first in order to have an app object [here](https://github.com/expressjs/express/blob/9e3dbb43744620b8b490c61c9269a168e0c0e3ae/lib/request.js#L219)

Reply:
we tried it together and apparently it works :shrug:

Reply:
![image](https://github.com/user-attachments/assets/918c20a3-71ea-47c9-a161-a6a21b6895ae)